lib/actionview_precompiler/render_parser.rb

```diff
@@ -117,7 +117,25 @@ def parse_render_from_options(options_hash)
       end
 
       render_type = (keys & render_type_keys)[0]
-      template = parse_str(options_hash[render_type])
+
+      node = options_hash[render_type]
+      if node.string?
+        template = node.to_string
+      elsif node.variable_reference?
+        dependency = node.variable_name.sub(/\A(?:\$|@{1,2})/, "")
+        template = "#{dependency.pluralize}/#{dependency.singularize}"
+        object_template = true
+      elsif node.vcall?
+        # TODO: do we need the sub?
+        dependency = node.variable_name.sub(/\A(?:\$|@{1,2})/, "")
+        object_template = true
+        template = "#{dependency.pluralize}/#{dependency.singularize}"
+      elsif node.call?
+        dependency = node.call_method_name
+        object_template = true
+        template = "#{dependency.pluralize}/#{dependency.singularize}"
```

Comment:
Maybe we can extract this to a method to avoid duplication? Say, `template_name_from_ident` or something similar?

```
def template_name_from_ident(name)
  "#{name.pluralize}/#{name.singularize}"
end
```